java/ql/lib/semmle/code/java/ControlFlowGraph.qll

```diff
@@ -84,45 +84,113 @@ private import Completion
 private import controlflow.internal.Preconditions
 private import controlflow.internal.SwitchCases
 
-/** A node in the expression-level control-flow graph. */
-class ControlFlowNode extends Top, @exprparent {
-  /** Gets the statement containing this node, if any. */
-  Stmt getEnclosingStmt() {
-    result = this or
-    result = this.(Expr).getEnclosingStmt()
+module ControlFlow {
+  private predicate hasControlFlow(Expr e) {
+    not e.getEnclosingStmt() instanceof ConstCase and
```

Comment:
This is interesting..

Reply:
In a line like `case "a" -> null;` I think this is trying to exclude the `"a"` but is accidentally excluding the `null` as well. I guess this wasn't an issue until switch expressions got added.

Reply:
Or maybe it's the arrow syntax that caused the problem, as extra expressions are now part of this statement.